src/argosim/metrics_utils.py

```diff
@@ -98,3 +98,161 @@ def compute_metrics(img1, img2):
         # "ssim": ssim(img1, img2),
     }
     return metrics
+
+
+def mask_main_lobe_elliptical(beam, fit_result, scale=1.0):
+    """
+    Apply an elliptical mask to suppress the main lobe from a beam image.
+
+    Parameters
+    ----------
+    beam : np.ndarray
+        2D beam image.
+    fit_result : dict
+        Dictionary containing the ellipse parameters (center, width, height, angle_deg).
+    scale : float
+        Scale factor to enlarge or shrink the ellipse mask (default is 1.0).
+
+    Returns
+    -------
+    beam_masked : np.ndarray
+        Beam image with the main lobe masked (set to 0).
+    """
+    center_x, center_y = fit_result["center"]
+    width = fit_result["width"] * scale
+    height = fit_result["height"] * scale
+    angle = np.radians(fit_result["angle_deg"])
+
+    y, x = np.indices(beam.shape)
+    x_shifted = x - center_x
+    y_shifted = y - center_y
+
+    cos_angle = np.cos(angle)
+    sin_angle = np.sin(angle)
+    x_rot = cos_angle * x_shifted + sin_angle * y_shifted
+    y_rot = -sin_angle * x_shifted + cos_angle * y_shifted
+
+    mask = (x_rot**2 / (width / 2) ** 2 + y_rot**2 / (height / 2) ** 2) < 1
+
+    beam_copy = beam.copy()
+    beam_copy[mask] = 0
+    return beam_copy
+
+
+def compute_sll(beam, fit_result, scale=1.0):
+    """
+    Compute the Side-Lobe Level (SLL) of a beam using an elliptical mask.
+
+    Parameters
+    ----------
+    beam : np.ndarray
+        The beam image (2D).
+    fit_result : dict
+        Elliptical fit result containing center, width, height, angle_deg.
+    scale : float
+        Scale factor for the ellipse mask.
+
+    Returns
+    -------
+    sll_db : float
+        Side-Lobe Level in dB.
+    """
+    main_lobe_peak = np.max(np.abs(beam))
+    beam_masked = mask_main_lobe_elliptical(np.abs(beam), fit_result, scale=scale)
+    side_lobe_peak = np.max(beam_masked)
+
+    sll_db = 10 * np.log10(side_lobe_peak / main_lobe_peak + 1e-12)
+    return sll_db
+
+
+def compute_fwhm(sigma_x, sigma_y):
+    """
+    Convert standard deviations to FWHM.
+
+    Parameters
+    ----------
+    sigma_x : float
+        Standard deviation along x-axis.
+    sigma_y : float
+        Standard deviation along y-axis.
+    Returns
+    -------
+    tuple
+        FWHM along x and y.
+    """
+    factor = 2.553  # approximation for FWHM = 2*sqrt(2*ln(2))
+    return factor * sigma_x, factor * sigma_y
+
+
+def compute_eccentricity(fwhm_x, fwhm_y):
+    """
+    Compute the eccentricity of an ellipse from FWHM values.
+
+    Parameters
+    ----------
+    fwhm_x : float
+        FWHM along x-axis.
+    fwhm_y : float
+        FWHM along y-axis.
+
+    Returns
+    -------
+    float
+        Ellipticity e ∈ [0, 1], where 0 = circle and 1 = highly elongated.
+    """
+    a = max(fwhm_x, fwhm_y)
```

Comment:
In principle the semi-major and minor axis are half the FWHM sizes. In the calculation of the eccentricity this factor cancels out because we compute b/a.